Objective-C/CBLReplicator.mm

```diff
@@ -730,26 +772,43 @@ static bool pushFilter(C4CollectionSpec collectionSpec,
                        C4String docID, C4String revID, C4RevisionFlags flags,
                        FLDict flbody, void *context) {
     auto replicator = (__bridge CBLReplicator*)context;
-    return [replicator filterDocument: docID revID: revID flags: flags
-                                 body: flbody pushing: true];
+    return [replicator filterDocument: collectionSpec docID: docID revID: revID
+                                flags: flags body: flbody pushing: true];
 }
 
 static bool pullFilter(C4CollectionSpec collectionSpec,
                        C4String docID, C4String revID, C4RevisionFlags flags,
                        FLDict flbody, void *context) {
     auto replicator = (__bridge CBLReplicator*)context;
-    return [replicator filterDocument: docID revID: revID flags: flags
-                                 body: flbody pushing: false];
+    return [replicator filterDocument: collectionSpec docID: docID revID: revID
+                                flags: flags body: flbody pushing: false];
 }
 
-- (bool) filterDocument: (C4String)docID
+- (bool) filterDocument: (C4CollectionSpec)c4spec
+                  docID: (C4String)docID
                   revID: (C4String)revID
                   flags: (C4RevisionFlags)flags
                    body: (FLDict)body
                 pushing: (bool)pushing
 {
-    CBLCollection* c = [_config.database defaultCollectionOrThrow];
-    auto doc = [[CBLDocument alloc] initWithCollection: c
+    CBLCollection* collection = nil;
+    if (c4spec.name.buf) {
+        NSError* error = nil;
+        NSString* name = slice2string(c4spec.name);
+        NSString* scopeName = slice2string(c4spec.scope);
+        collection = [_config.database collectionWithName: name
```

Comment:
This should be gotten from the _config.collectionConfigs. For the conflict resolver, I suggested to set callbackContext in C4ReplicationCollection with the Collection object. However, when setting that the context for the filter could be the collection object as well which might not be working here (check with LiteCore or test it yourself). 

If there is conflict of setting the context object, another option would be lazily creating a dictionary of full collection name (< scope > . < collection >) and the CBLCollection object out of _config.collectionConfig in order to get to the CBLCollection object form the c4spec.

Reply:
Setting C4ReplicationCollection is not making any difference. `onDocumentsEnded`, is still returning the replicator we set at 'C4ReplicatorParameters.callbackContext'. I guess, that param is not used currently.